fortran/src/cudafor.f90

```diff
@@ -70,45 +70,29 @@ end function cudaSetDevice
 end interface cudaSetDevice
 !End cudaSetDevice
 
-!Start cudaMAlloc
-interface cudaMAlloc
-integer(c_int) function cudaMAlloc(devPtr, size) bind(c, name = "cudaMalloc")
+!Start cudaMalloc
+interface cudaMalloc
+integer(c_int) function cudaMalloc(devPtr, size) bind(c, name = "cudaMalloc")
 import :: c_int, c_size_t
 import :: c_devptr
 implicit none
 type(c_devptr) :: devPtr
 integer(c_size_t), value :: size
-end function cudaMAlloc
-end interface cudaMAlloc
-!End cudaMAlloc
+end function cudaMalloc
+end interface cudaMalloc
+!End cudaMalloc
 
 !Start cudaMemCpy
 interface cudaMemCpy
```

Comment:
I still don't understand the point of the D2H/H2D calls. Why not just do it the way I did, which is map cudaMemcpy (Fortran) to .. cudaMemcpy (C). What it the issue here that you are trying to solve ?

In my patch I just map one to the other and everything seems to work properly.